service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/GeneratedOrderControllerTest.java

```diff
@@ -105,30 +105,60 @@ void aboutToStartShouldReturnErrorsWhenFamilymanNumberIsNotProvided() throws Exc
     void midEventShouldGenerateOrderDocument() throws Exception {
         byte[] pdf = {1, 2, 3, 4, 5};
         Document document = document();
-        DocmosisDocument docmosisDocument = new DocmosisDocument(C21.getDocumentTitle(), pdf);
+        DocmosisDocument docmosisDocument = new DocmosisDocument("order.pdf", pdf);
 
         given(docmosisDocumentGeneratorService.generateDocmosisDocument(any(), any())).willReturn(docmosisDocument);
-        given(uploadDocumentService.uploadPDF(USER_ID, AUTH_TOKEN, pdf, C21.getDocumentTitle()))
+        given(uploadDocumentService.uploadPDF(USER_ID, AUTH_TOKEN, pdf, "blank_order_c21.pdf"))
             .willReturn(document);
 
         AboutToStartOrSubmitCallbackResponse callbackResponse = makeRequest(callbackRequest(), "mid-event");
 
         CaseData caseData = mapper.convertValue(callbackResponse.getData(), CaseData.class);
 
-        assertThat(caseData.getOrder().getDocument()).isEqualTo(DocumentReference.builder()
+        assertThat(caseData.getOrderTypeAndDocument().getDocument()).isEqualTo(DocumentReference.builder()
             .binaryUrl(document.links.binary.href)
             .filename(document.originalDocumentName)
             .url(document.links.self.href)
             .build());
     }
 
     @Test
-    void aboutToSubmitShouldUpdateCaseData() throws Exception {
+    void aboutToSubmitShouldUpdateCaseDataAccordinglyWhenC21IsSelected() throws Exception {
         AboutToStartOrSubmitCallbackResponse callbackResponse = makeRequest(callbackRequest(), "about-to-submit");
 
         CaseData caseData = mapper.convertValue(callbackResponse.getData(), CaseData.class);
 
-        GeneratedOrder expectedOrder = GeneratedOrder.builder()
+        GeneratedOrder expectedC21Order = buildExpectedC21Order();
+        aboutToSubmitAssertions(caseData, expectedC21Order);
+    }
+
+    @Test
+    void aboutToSubmitShouldUpdateCaseDataAccordinglyWhenCareOrderIsSelected() throws Exception {
+        AboutToStartOrSubmitCallbackResponse callbackResponse = makeRequest(careOrderRequest(), "about-to-submit");
+
+        CaseData caseData = mapper.convertValue(callbackResponse.getData(), CaseData.class);
+
+        GeneratedOrder expectedCareOrder = buildExpectedCareOrder();
+        aboutToSubmitAssertions(caseData, expectedCareOrder);
+    }
+
+    @Test
+    void shouldTriggerOrderEventWhenSubmitted() throws Exception {
+        String expectedCaseReference = "19898989";
+        makeSubmittedRequest(buildCallbackRequest());
+
+        verify(notificationClient, times(1)).sendEmail(
+            eq(ORDER_NOTIFICATION_TEMPLATE), eq(LOCAL_AUTHORITY_EMAIL_ADDRESS),
+            eq(expectedOrderLocalAuthorityParameters()), eq(expectedCaseReference));
+    }
+
+    private GeneratedOrder buildExpectedC21Order() {
```

Comment:
GitHub is being silly and not letting me see full buildExpectedC21Order and buildExpectedCareOrder but are these two not largely the same? Could we not just have the one method with a param of GeneratedOrderType?

Reply:
They are not exactly the same so I'd rather keep them separate for now - if we have enough common fields later when we have multiple orders, can extract common parts out into private methods